src/file_handle/web.rs

```diff
@@ -1,15 +1,31 @@
+use crate::file_dialog::FileDialog;
 use wasm_bindgen::prelude::*;
 use wasm_bindgen::JsCast;
 
-pub struct FileHandle(web_sys::File);
+#[derive(Clone, Debug)]
+pub(crate) enum WasmFileHandleKind {
+    Readable(web_sys::File),
+    Writable(FileDialog),
+}
+
+pub struct FileHandle(pub(crate) WasmFileHandleKind);
 
 impl FileHandle {
+    /// Wrap a [`web_sys::File`] for reading. Use with [`FileHandle::read`]
     pub fn wrap(file: web_sys::File) -> Self {
-        Self(file)
+        Self(WasmFileHandleKind::Readable(file))
+    }
+
+    /// Create a dummy `FileHandle`. Use with [`FileHandle::write`].
+    pub fn writable(dialog: FileDialog) -> Self {
+        FileHandle(WasmFileHandleKind::Writable(dialog))
     }
```

Comment:
What's the purpose of this one?

tbh I'm not even sure why I made regular wrap public, so if there is no specific reason I would refrain from adding another, especially platform specific one

Reply:
It is used here `src\backend\wasm\file_dialog.rs:13:32`.
I made it `pub` to match the other `wrap`.
I can make it `pub(crate)`

Reply:
pub(crate) would be good, I think I will change regular wraps to that as well, unless I find a reason why those are public :smile:

Reply:
Fix: [c0d0ba5](https://github.com/PolyMeilex/rfd/pull/134/commits/c0d0ba5bb2ddc7f1c291c6c7c2d3c7ebf7732f8d)

Obviously PR is now a major version change if it wasn't already.